src/components/Authentication/LoginPage.js

```diff
@@ -56,22 +75,44 @@ const LoginPage = () => {
     const [getUserType] = useLazyQuery(GET_USER_TYPE, {
         variables: { username: email },
         onCompleted: (data) => {
-            setUserType(data.userType);
+            setUserType(data?.userType?.userType);
+            setGoogleAuthEnabled(data?.userType?.googleAuthEnabled);
             if (userType === null) {
                 setHasError(true);
             }
         },
     });
+    const [getVerifyGoogleOauthTokenStatus] = useLazyQuery(
+        VERIFY_GOOGLE_OAUTH,
+        {
+            variables: { loginEmail: email, oauthEmail: googleAuthEmail },
+            onCompleted: async (data) => {
+                setVerifyGoogleOauthTokenStatus(
+                    data?.verifyGoogleOauth?.verified
+                );
+
+                if (data?.verifyGoogleOauth?.verified) {
+                    dispatch(
+                        await setToken(
+                            data.verifyGoogleOauth.token,
+                            true,
+                            email
+                        )
+                    );
+                    history.push('/');
+                }
+                if (verifyGoogleOauthTokenStatus === null) {
```

Comment:
the ordering of this is a bit odd. you update the state variable on line 90, but it won't get updated until the next render, so this line will never really do anything. if you want to do error handling based on the `verifyGoogleOauthTokenStatus` then check for this in the beginning. 
```
if( data.verifyGoogleOauth === null )
```